src/Macro.php

```diff
@@ -24,14 +24,39 @@ public function __construct(
         $interfaces = array()
     ) {
         parent::__construct($method, $alias, $class, $methodName, $interfaces);
+
+        // Remove the static tag
+        $this->phpdoc->deleteTag(
+            $this->phpdoc->getTagsByName('static')[0]
+        );
     }
 
     /**
      * @param \ReflectionFunctionAbstract $method
      */
     protected function initPhpDoc($method)
     {
-        $this->phpdoc = new DocBlock($method);
+        // Init DocBlock only with a description
+        $this->phpdoc = new DocBlock('/** No description available. */');
+
+        // Add macro parameters
+        foreach ($method->getParameters() as $parameter) {
+            $type = $parameter->hasType() ? $parameter->getType() : 'mixed';
+            $type .= $parameter->hasType() && $parameter->getType()->allowsNull() ? '|null' : '';
+
+            $name = $parameter->isVariadic() ? '...' : '';
+            $name .= '$' . $parameter->getName();
+
+            $this->phpdoc->appendTag(Tag::createInstance("@param {$type} {$name}"));
+        }
+
+        // Add macro return type
+        if ($method->hasReturnType()) {
+            $type = $method->getReturnType();
```

Comment:
Here I have the same issue:
```
$ ./artisan ide-helper:generate

   ErrorException

  Function ReflectionType::__toString() is deprecated

  at vendor/barryvdh/laravel-ide-helper/src/Macro.php:56
    52|
    53|         // Add macro return type
    54|         if ($method->hasReturnType()) {
    55|             $type = $method->getReturnType();
  > 56|             $type .= $method->getReturnType()->allowsNull() ? '|null' : '';
    57|
    58|             $this->phpdoc->appendTag(Tag::createInstance("@return {$type}"));
    59|         }
    60|     }

      +21 vendor frames
  22  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
```
```suggestion
            $type = $method->getReturnType()->getName();
```

Reply:
I have changed this to your suggestion.